src/nullablevector.jl

```diff
@@ -311,3 +311,14 @@ function Base.empty!(X::NullableVector)
     empty!(X.isnull)
     return X
 end
+
+notnullarray = Union{filter!(x -> !isa(x, Type{NullableArray}), subtypes(AbstractArray))...}
+notnullmatrix = typeintersect(notnullarray, AbstractMatrix)
+notnullvector = typeintersect(notnullarray, AbstractVector)
+
+Base.vcat{T <: notnullarray}(A::T, B::NullableArray) = Base.vcat(NullableArray(A), B)
```

Comment:
I think this case of adding an empty NullableArray to the beginning of the `vcat/hcat` call only works on `vcat` with `Vector`s, but I've added it for that case. Every other case I tried will fail in dimension mismatch checks